ce/api/geo.py

```diff
@@ -1,6 +1,101 @@
+import logging
+import collections
+import sys
+from collections import OrderedDict
+from threading import RLock
+
 import numpy as np
 import numpy.ma as ma
 from shapely.geometry import Polygon, Point, CAP_STYLE
+from shapely.wkt import loads
+
+
+### From http://stackoverflow.com/a/30316760/597593
+from numbers import Number
+from collections import Set, Mapping, deque
+try: # Python 2
+    zero_depth_bases = (basestring, Number, xrange, bytearray)
+    iteritems = 'iteritems'
+except NameError: # Python 3
+    zero_depth_bases = (str, bytes, Number, range, bytearray)
+    iteritems = 'items'
+
+def getsize(obj):
+    """Recursively iterate to sum size of object & members."""
+    def inner(obj, _seen_ids = set()):
+        obj_id = id(obj)
+        if obj_id in _seen_ids:
+            return 0
+        _seen_ids.add(obj_id)
+        size = sys.getsizeof(obj)
+        if isinstance(obj, zero_depth_bases):
+            pass # bypass remaining control flow and return
+        elif isinstance(obj, (tuple, list, Set, deque)):
+            size += sum(inner(i) for i in obj)
+        elif isinstance(obj, Mapping) or hasattr(obj, iteritems):
+            size += sum(inner(k) + inner(v) for k, v in getattr(obj, iteritems)())
+        # Now assume custom object instances
+        elif hasattr(obj, '__slots__'):
+            size += sum(inner(getattr(obj, s)) for s in obj.__slots__ if hasattr(obj, s))
+        else:
+            attr = getattr(obj, '__dict__', None)
+            if attr is not None:
+                size += inner(attr)
+        return size
+    return inner(obj)
+### End from http://stackoverflow.com/a/30316760/597593
+
+log = logging.getLogger(__name__)
+cache_lock = RLock()
+
+class memoize_mask(object):
+    '''
+    Decorator. Caches wktToMask keyed to model_id and the WKT string
+    '''
+    def __init__(self, func, maxsize=50):
```

Comment:
In my tests, each addition to the cache added ~9k to the cache so ~5800 requests at 50 MB. Probably too large with current data, but this will be affected by different grid resolutions. I would think enough space to cache each unique_id with 5 polygons would be sufficient.

Initially the max was enforced to 100 requests by default and then later changed to a memory size limit.